product-service/handlers/getProductsList.ts

```diff
@@ -1,18 +1,40 @@
+import {Pool} from "pg";
 import {APIGatewayProxyHandler} from "aws-lambda";
-import products from "../products";
-import headers from "./headers";
+import {headers, internalError} from "./helpers";
+import {DB_OPTIONS} from "./db";
 
-export const getProductsList: APIGatewayProxyHandler = async () => {
-    return {
-        statusCode: 200,
-        headers,
-        body: JSON.stringify(
-            products.map((item) => ({
-                ...item,
-                image: `https://source.unsplash.com/random?sig=${Math.random()}`,
-            })),
-            null,
-            2
-        ),
-    };
+// cached variable
+let pool
+
+export const getProductsList: APIGatewayProxyHandler = async (event) => {
+    console.log(event)
+    // Check if pool already exists. Create new one if not.
+    if (!pool) {
+        pool = new Pool(DB_OPTIONS);
+    }
+
+    // Get client from pool
+    const client = await pool.connect();
+    console.log(client)
+
+    try {
+        const {rows: products} = await client.query(`select * from products p inner join stocks s on s.product_id = p.id`);
```

Comment:
is it correct?

Reply:
I'm not so good at sql but what's wrong about it? I got my products and just get **count** from stocks table with `inner join`